textile/features.textile

```diff
@@ -899,6 +909,27 @@ h3(#backoff-jitter). Incremental backoff and jitter
 h3(#realtime-compatibility). Forwards compatibility
 * @(RTF1)@ The library must apply the "robustness principle":https://en.wikipedia.org/wiki/Robustness_principle in its processing of requests and responses with the Ably system. In particular, deserialization of ProtocolMessages and related types, and associated enums, must be tolerant to unrecognised attributes or enum values. Such unrecognised values must be ignored.
 
+h3(#wrapper-sdk-proxy). Wrapper SDK proxy client
+
+A _wrapper SDK_ is an Ably-authored non-core SDK.
+
+The core SDK provides an API for wrapper SDKs to supply Ably with analytics information that allows us track the usage of these SDKs.
```

Comment:
Something that we should perhaps consider here - right now, connections to the server are made by the underlying SDK, and because `autoconnect=true` we never see connections from the wrapper SDKs.

Obviously with autoconnect, that's not something we can realistically do when the option is turned on. But perhaps we should decide if we want to provide wrapper details in connections in any way, and also specify it?

Reply:
So, in [my assumptions of the requirements in ADR-117](https://ably.atlassian.net/wiki/spaces/ENG/pages/3413016589/ADR-117+Tracking+wrapper+SDK+usage+continued#Requirements) I assumed that we did not want to attribute Realtime connection events to a wrapper SDK. This was based on the assumption that `autoconnect` would be used, though. But, if you look at [Matt's response to my question about what we'd like to track](https://ably.atlassian.net/wiki/spaces/ENG/pages/3413016589/ADR-117+Tracking+wrapper+SDK+usage+continued#Which-REST-requests-do-we-want-to-be-able-to-attribute-to-a-modern-wrapper-SDK%3F), he seems to believe that connections are not something we are interested in tracking.

Reply:
Ah perfect, that's already been discussed :) Yeah I agree attachments are more valuable than just connections.

Reply:
I think tracking connections is fundamentally incorrect in terms of analytics. The connection is initiated by `RealtimeClient`, so it shouldn’t be counted as Chat SDK usage. Otherwise, the wrapper SDK usage will obscure all `RealtimeClient` usage, leading to incorrect assumptions

Reply:
> I think tracking connections is fundamentally incorrect in terms of analytics. The connection is initiated by `RealtimeClient`, so it shouldn’t be counted as Chat SDK usage. Otherwise, the wrapper SDK usage will obscure all `RealtimeClient` usage, leading to incorrect assumptions

Yeah exactly, all of our current chat analytics are based on attachments + frontend API requests anyway

Reply:
Can we resolve this conversation?

Reply:
Yes